sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -530,13 +568,16 @@ private async Task RunBatchReceiveLoopAsync(CancellationToken cancellationToken)
                 catch (Exception ex)
                 {
                     // Log another exception
-                    _logger.LogError(ex, $"An unhandled exception occurred in the message batch receive loop ({_details.Value})");
+                    _logger.LogError(ex, $"An unhandled exception occurred in the message batch receive loop ({_details.Value}). If a minimum batch size was set, any cached session messages will be abandoned.");
 
                     if (_isSessionsEnabled && receiver != null)
                     {
                         // Attempt to close the session and release session lock to accept a new session on the next loop iteration
                         try
                         {
+                            // Abandon any remaining session messages, they can be received again in the next iteration if the same session is accepted.
+                            await TryAbandonRemainingSessionMessages(receiver, cancellationTokenSource.Token).ConfigureAwait(false);
```

Comment:
nit: I don't think this is necessary as the lock is at the session level. Abandoning a session message would only serve to allow the same message to be delivered on the already locked session. Since we are closing the session, I don't think it will have any impact.

Reply:
Ok that makes sense. I can change this to instead drop all the messages in the cache.

Reply:
Well, actually when it creates a new session in the receive loop it will drop them anyway.